libcore/File.vala

```diff
@@ -624,18 +626,126 @@ public class Files.File : GLib.Object {
         }
     }
 
-    public void update_icon (int size, int scale) {
-        if (size <= 1) {
-            return;
+    /* Icon/thumnail related functions */
+
+    public GLib.Icon? get_icon_user_special_dirs (string path) {
```

Comment:
Isn't this covered by `FileAttribute.STANDARD_ICON` in FileInfo?

Reply:
Thanks for taking a look at this. You are probably right - I'll look into it.  The scope of this PR seems to have grown substantially, into an extensive review of icon handling as fixing the issue proved more complicated than envisaged.  A lot of the code is old and hard to understand. Simplifying/clarifying will aid future porting to Gtk4.

Reply:
If its OK, I would rather leave changing this (if required) to a later PR.  The scope of this one is already larger than intended.